### ISSUE TITLE
Brain Challenge Regression

### ISSUE THREAD

**Issue by thundergun1** [hikari-no-yume/touchHLE#329]
### App DB report link

https://appdb.touchhle.org/apps/22

### App name

Brain Challenge

### App version

1.0

### touchHLE version

v0.2.2-104-gbd0ad53

### Operating system

Android 12

### GPU

Adreno

### What is the problem with the app?

While not reported on the appdb, Brain Challenge has been working at a 4-star level since v0.2.2 (as well as the Dungeon Hunter, Crystal Defenders and Zombie Infection updates) but as of last night's Gangstar update, Brain Challenge is now crashing at the 'How About You?' Intro screen just like before 0.2.2.

### If touchHLE crashed, please paste the log output.

```text
touchHLE v0.2.2-104-gbd0ad53 — https://touchhle.org/

touchHLE: Base path for touchHLE files: /storage/emulated/0/Android/data/org.touchhle.android/files
No app specified, opening app picker. Use the --help flag to see command-line usage.
touchHLE::window: Accelerometer detected: lsm6dsm Accelerometer Non-wakeup.
touchHLE::gles: Creating an OpenGL ES 1.1 context:
touchHLE::gles: Trying: Native OpenGL ES 1.1
touchHLE::gles: => Success!
touchHLE::window: Driver info: OpenGL ES-CM 1.1 / Qualcomm / Adreno (TM) 619
Picked: /storage/emulated/0/Android/data/org.touchhle.android/files/touchHLE_apps/Brain Challenge.ipa
App bundle info:
- Display name: BrainChal.
- Version: 1.0
- Identifier: com.gameloft.BrainChallenge
- Internal name (canonical): BrainChallenge.app
- Minimum OS version: (not specified)

No options found for this app in touchHLE_default_options.txt
No options found for this app in /storage/emulated/0/Android/data/org.touchhle.android/files/touchHLE_options.txt

touchHLE::window: Accelerometer detected: lsm6dsm Accelerometer Non-wakeup.
touchHLE::gles: Creating an OpenGL ES 1.1 context:
touchHLE::gles: Trying: Native OpenGL ES 1.1
touchHLE::gles: => Success!
touchHLE::window: Driver info: OpenGL ES-CM 1.1 / Qualcomm / Adreno (TM) 619
touchHLE::mach_o: Loading armv6 slice for "BrainChallenge"
touchHLE::mach_o: Loading armv6 slice for "libstdc++.6.dylib"
touchHLE::mach_o: Loading armv6 slice for "libgcc_s.1.dylib"
touchHLE::dyld: Warning: unhandled non-lazy symbol "___mb_cur_max" at 0x3f48a654 in "libstdc++.6.dylib"
CPU emulation begins now.
touchHLE::libc::cxxabi: TODO: __cxa_atexit(GuestFunction(0x374ef884), 0x0, 0x3748a000) (unimplemented)
touchHLE::libc::cxxabi: TODO: __cxa_atexit(GuestFunction(0x374ef8b8), 0x0, 0x3748a000) (unimplemented)
touchHLE::libc::cxxabi: TODO: __cxa_atexit(GuestFunction(0x2090), 0xfffffe1c, 0xfffffe1c) (unimplemented)
touchHLE::libc::cxxabi: TODO: __cxa_finalize(0xfffffe1c) (unimplemented)
touchHLE::libc::cxxabi: TODO: __cxa_finalize(0xfffffe1c) (unimplemented)
touchHLE::libc::cxxabi: TODO: __cxa_atexit(GuestFunction(0x2544), 0x374f7d00, 0x1000) (unimplemented)
touchHLE::libc::cxxabi: TODO: __cxa_atexit(GuestFunction(0x2544), 0x374f7d10, 0x1000) (unimplemented)
touchHLE::libc::cxxabi: TODO: __cxa_atexit(GuestFunction(0x2544), 0x374f7d20, 0x1000) (unimplemented)
touchHLE::libc::cxxabi: TODO: __cxa_atexit(GuestFunction(0x2544), 0x374f7d30, 0x1000) (unimplemented)
touchHLE::libc::cxxabi: TODO: __cxa_atexit(GuestFunction(0x2544), 0x374f7d40, 0x1000) (unimplemented)
touchHLE::libc::cxxabi: TODO: __cxa_atexit(GuestFunction(0x2544), 0x374f7d50, 0x1000) (unimplemented)
touchHLE::libc::cxxabi: TODO: __cxa_atexit(GuestFunction(0x2544), 0x374f7d60, 0x1000) (unimplemented)
touchHLE::libc::cxxabi: TODO: __cxa_atexit(GuestFunction(0x2544), 0x374f7d70, 0x1000) (unimplemented)
touchHLE::libc::cxxabi: TODO: __cxa_atexit(GuestFunction(0x2544), 0x374f7d80, 0x1000) (unimplemented)
touchHLE::libc::cxxabi: TODO: __cxa_atexit(GuestFunction(0x2544), 0x374f7d90, 0x1000) (unimplemented)
touchHLE::libc::cxxabi: TODO: __cxa_atexit(GuestFunction(0x2544), 0x374f7da0, 0x1000) (unimplemented)
touchHLE::libc::cxxabi: TODO: __cxa_atexit(GuestFunction(0x2544), 0x374f7db0, 0x1000) (unimplemented)
touchHLE::libc::cxxabi: TODO: __cxa_atexit(GuestFunction(0x2544), 0x374f7dc0, 0x1000) (unimplemented)
touchHLE::libc::cxxabi: TODO: __cxa_atexit(GuestFunction(0x2544), 0x374f7dd0, 0x1000) (unimplemented)
touchHLE::libc::cxxabi: TODO: __cxa_atexit(GuestFunction(0x2544), 0x374f7de0, 0x1000) (unimplemented)
touchHLE::libc::cxxabi: TODO: __cxa_atexit(GuestFunction(0x2544), 0x374f7df0, 0x1000) (unimplemented)
touchHLE::libc::cxxabi: TODO: __cxa_atexit(GuestFunction(0x2544), 0x374f7e00, 0x1000) (unimplemented)
touchHLE::libc::cxxabi: TODO: __cxa_atexit(GuestFunction(0x2544), 0x374f7e10, 0x1000) (unimplemented)
touchHLE::frameworks::uikit::ui_nib: TODO: UIProxyObject replacement for IBFirstResponder, instance 0x374f8550 left unreplaced
touchHLE::gles: Creating an OpenGL ES 1.1 context:
touchHLE::gles: Trying: Native OpenGL ES 1.1
touchHLE::gles: => Success!
touchHLE::frameworks::opengles::eagl: Driver info: OpenGL ES-CM 1.1 / Qualcomm / Adreno (TM) 619
touchHLE::frameworks::uikit::ui_view::ui_control::ui_text_field: TODO: setAutocorrectionType:1
touchHLE::frameworks::uikit::ui_view::ui_control::ui_text_field: TODO: setKeyboardType:6
touchHLE::libc::posix_io: Warning: read(3, 0x374fcb10, 0x4d765) read only 0x4d764 bytes
touchHLE::libc::posix_io: Warning: read(3, 0x37597a70, 0x45c72a) read only 0x45c729 bytes
touchHLE::libc::posix_io: Warning: read(3, 0x37e508d0, 0xa42c5) read only 0xa42c4 bytes
touchHLE::window: This app uses the accelerometer.
touchHLE::window: Your device's accelerometer will be used for accelerometer simulation.
touchHLE::window: Connect a controller if you would prefer to use an analog stick.
touchHLE::libc::posix_io: Warning: read(3, 0x30009000, 0x4dc004) read only 0x4dc003 bytes
touchHLE::libc::posix_io: Warning: read(3, 0x376a6130, 0xd196) read only 0xd195 bytes
touchHLE::libc::posix_io: Warning: read(3, 0x377b8860, 0xa42c5) read only 0xa42c4 bytes
touchHLE::libc::posix_io: Warning: read(3, 0x377b8860, 0x2e254) read only 0x2e253 bytes
touchHLE::frameworks::foundation::ns_locale: The app requested your preferred languages. ["en"] will be reported based on your system language preferences.
touchHLE::libc::posix_io: Warning: read(3, 0x3785cb30, 0xa42c5) read only 0xa42c4 bytes
touchHLE::frameworks::media_player::movie_player: TODO: [(MPMoviePlayerController*)0x3750da10 initWithContentURL:0x3750da00 (GuestPathBuf("/var/mobile/Applications/00000000-0000-0000-0000-000000000000/BrainChallenge.app/brain.mp4"))]
touchHLE::frameworks::media_player::movie_player: TODO: [(MPMoviePlayerController*)0x3750da10 play]
Panic at src/objc/messages.rs:44:5: assertion failed: orig_class != nil
Register state immediately after panic:
	 R0: 0x3750d9c0	 R1: 0x00072570	 R2: 0x00000000	 R3: 0x00000000
	 R4: 0x00000001	 R5: 0x374f7e20	 R6: 0xffffff20	 R7: 0xffffff10
	 R8: 0x00000000	 R9: 0x3f490ed0	R10: 0x00000000	R11: 0x00000000
	R12: 0x0007df58	 SP: 0xffffff04	 LR: 0x000642cf	 PC: 0x0006de2c
Attempting to produce stack trace for main thread:
 0. 0x6de2c (PC)
 1. 0x642cf (LR)
 2. 0x2034
Next FP (0x0) is outside the stack.
```

### By submitting this issue, I certify that…

- [X] I have tested that this version of this app is at least partially working in a previous version of touchHLE.
- [X] I understand that linking to, discussing or asking for pirated content is not permitted, and will lead to an immediate ban.
- [X] I understand that “Abandonware” and illegal copies of free games are also pirated content.

**Comment by ciciplusplus** [comment]
Hi, appdb clearly indicates a 2 star game, the full support for it was
never claimed.
In which version of the touchHLE it was working for you before?

**Comment by thundergun1** [comment]
It was working from the v0.2.2 release and any update up until v0.2.2-104-gbd0ad53 from last night. I have been daily driving it so I can confirm it works I just never got round to putting it on the appdb

**Comment by hikari-no-yume** [comment]
I can't reproduce this. With version 1.0 of the game, it crashes on latest trunk after the language select, and it's the same for v0.2.2. Can you provide a log and screenshot of it working with some other version of touchHLE?

**Comment by thundergun1** [comment]
I think I've realised the issue. I believe I completed first boot on another fork which bypassed the crash a while ago. From then on my save file worked on the trunk and other builds. I lost my save file last night and so can no longer bypass the first boot. I just reinstalled the Zombie Infection update which I had definitely played Brain Challenge on but it crashes now so it must have been another fork bypassing the first boot. Apologies for this, i'll close the issue.
